tests/unit_tests/test_ustar_cp/test_cpdAssignUStarTh20100901.py

```diff
@@ -0,0 +1,100 @@
+import pytest
+import matlab.engine
+import numpy as np
+
+@pytest.fixture
+def mock_stats():
+    # Create a mock Stats structure
+    class MockStats:
+        def __init__(self):
+            self.mt = np.random.rand(40)
+            self.Cp = np.random.rand(40)
+            self.b1 = np.random.rand(40)
+            self.c2 = np.random.rand(40)
+            self.cib1 = np.random.rand(40)
+            self.cic2 = np.random.rand(40)
+            self.p = np.random.rand(40)
+
+    return MockStats()
+
+def test_cpdAssignUStarTh20100901_basic(matlab_engine, mock_stats):
+    # Convert mock stats to MATLAB structure
+    stats_matlab = matlab_engine.struct()
+    for field in ['mt', 'Cp', 'b1', 'c2', 'cib1', 'cic2', 'p']:
+        setattr(stats_matlab, field, matlab.double(getattr(mock_stats, field).tolist()))
```

Comment:
This pattern doesn't seem right to me: a MATLAB struct seems to be represented as a dictionary so using `setattr` doesn't work for me with R2024a (24.1):
```
>>> import matlab.engine
>>> s = eng.struct()
>>> setattr(s, "field", 42)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'dict' object has no attribute 'field'
```
Is this a change with R2024b? Do we need to change our requirements?

Reply:
Fixed